[CS2113T-W09-1] Finux

tzexern opened the pull request:
Finux is a CLI Style application that allows the user to make better financial decisions based on the information recorded in the app. If you’re familiar with the CLI command interface, you will enjoy the benefit of speeding up your financial management rather than using the traditional management system.

8kdesign commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+1. **Verify the setup.**
+   1. After importing successfully, locate the `src/main/java/seedu.duke/Duke.java` file, right click it, 
+      and choose `Run...`. If the setup is correct, you should see the following:
+   ![Login_Page](img/LoginPageExampleOutput.png)
```
Perhaps the unused space on the right side can be cropped out so that you can have more space to make the image bigger.

marklowsk replied:
Yes, thank you, will change it!

brandonfoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+
+### 3.1 Architecture
+
+![ArchitectureDiagram](img/ArchitectureDiagram.png)\
```
Should the arrows coming out of the "Finux" box be connected to some of the other boxes?

marklowsk replied:
It is intended, Finux initializes all the other components.
Thanks for asking!

8kdesign commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+
+### 3.1 Architecture
+
+![ArchitectureDiagram](img/ArchitectureDiagram.png)\
```
Perhaps the color of the light blue arrow can be changed. It is a bit hard to see at the moment.

marklowsk replied:
Sure no problem!

aliciatay-zls commented on src/main/java/seedu/duke/Finux.java:
```diff
@@ -0,0 +1,85 @@
+            storage = new Storage();
+            parserHandler = new ParserHandler();
+            commandHandler = new CommandHandler();
+            storage.loadFile();
```
![image](https://user-images.githubusercontent.com/61925875/113076978-c6eddf80-9202-11eb-96f4-89af303ed38b.png)


Was the call to loadFile() of the Finux class intentionally omitted?
Should "ArrayList" be "ArrayList//<Record//>" for the returning arrow after the call to getRecordListData() of Storage?

marklowsk replied:
Thanks for pointing this out! We have changed it.

8kdesign commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+* At shut down: Shuts down the components and invokes cleanup method where necessary.
+
+The rest of the Application consists of six components.
+* `Ui`: The user interface (Ui) of the App which handles all user input and Application output.
```
Is there a reason why both "App" and "Application" are used here? If they refer to the same thing, it might be confusing for the reader.

marklowsk replied:
Thanks for pointing this out! Will change it.

brandonfoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+* the execution of commands to print the result of a `command`
+
+### 3.3 Parser Component
+![ParserHandlerClassDiagram](img/ParserHandlerClassDiagram.png) <br>
```
Looks a bit text-heavy, would it be possible to show only the most important methods of the 2 classes? I think it would be best to focus on the public methods, since they make up the API.

aliciatay-zls commented on src/main/java/seedu/duke/Finux.java:
```diff
@@ -0,0 +1,85 @@
+    /**
+     * Loops the application until an EXIT command is parsed.
+     */
+    private void commandLooper() {
+        while (!commandHandler.isExit()) {
+            String rawInput = ui.getUserInput();
+            ArrayList<String> parsedStringList = parserHandler.getParseInput(rawInput);
+            assert parsedStringList.size() != 0 : "Empty Parser Error";
+            Command command = commandHandler.parseCommand(parsedStringList, records);
+            if (command == null) {
+                continue;
+            }
+            command.execute(records, ui, storage, creditScoreMap);
+        }
+    }
```
![image](https://user-images.githubusercontent.com/61925875/113076858-85f5cb00-9202-11eb-8d1f-dac8a40453d0.png)


Should the lifeline end at the cross for each of the classes?

marklowsk replied:
PlantUML limitation, we have explained it in one of the sections.

brandonfoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+
+   Afterward, the checking mechanism will loop until no valid next options are left in the input before calling
+   `extractFinalPart`.
+   * Valid next option format: `' <option> '` with 1 leading and trailing whitespace. e.g. `' -e '`
```
Not your fault, but it seems like the github pages theme breaks code formatting here. Maybe you could force the code block into a new line using `\`.

brandonfoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+   * Valid next option format: `' <option> '` with 1 leading and trailing whitespace. e.g. `' -e '`
+4. `extractFinalPart` will check if the last trimmed input ends with option. If yes, extract the option and 
+   add an empty string as the argument, else just add the last trimmed input to the ArrayList. 
+5. Finally, after the extraction to ArrayList<String> is complete, `extractFinalPart` will 
```
Seems like the code blocks in this section are not being rendered properly. Also, should ArrayList be enclosed with code tags?

brandonfoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+   * During processing, next option should be in the form of `' <option> '` with 1 leading and trailing whitespace.
+     e.g. `' -e '`
+3. As rearrangement of options is allowed, option detection should cater to non-fixed option order. 
+   Apache Commons Lang, 3.11, providing the StringUtils class is used to cater to consideration 2.
```
Should code tags be used around StringUtils (same for the 3 bullet points below)?

8kdesign commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+sets the `isExit = true`, resulting in `Finux` proceeding to call `end()` to exit the Application.
+
+### 3.5 Command Component
+![CommandClassDiagram](img/CommandClassDiagram.png)
```
The image is too long so text appear too small. Would it be possible to shift some of the classes to a new line instead of arranging all of them in one line?

marklowsk replied:
PlantUML limitations, decided to split the diagram into 2 images.
Thanks!

brandonfoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+### 3.4 CommandHandler Component
+![CommandHandlerClassDiagram](img/CommandHandlerClassDiagram.png)\
+_Figure 6: CommandHandler Class Diagram_\
+![CommandHandlerSequenceDiagram](img/CommandHandlerSequenceDiagram.png)\
```
Should the Finux lifeline also contain a activation bar here?

marklowsk replied:
Right, will add it in, thanks!

isaharon commented on src/main/java/seedu/duke/Finux.java:
```diff
@@ -0,0 +1,85 @@
+            storage = new Storage();
+            parserHandler = new ParserHandler();
+            commandHandler = new CommandHandler();
+            storage.loadFile();
```
Should there be a call to the loadFile() method of Storage here? Also, the line for creditScoreMap seems to be omitted from the diagram as well. Was it probably intended?
![Presentation1](https://user-images.githubusercontent.com/19249948/113076601-00721b00-9202-11eb-8461-37303e2eb9cf.png)

marklowsk replied:
Thanks for pointing this out! We have changed it.

8kdesign commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+#### 4.3.2 Design Consideration
+...
+
+### 4.2 List Feature
+The `list` feature allows Finux users to list records that they have entered into the system.
+
+To list loans:
+
+`list -l`
+
+To list expenses:
+
+`list -e`
+
+To list savings:
+
+`list -s`
```
For this part, perhaps you might want to remove it if you have nothing to share about its implementation.

marklowsk replied:
I think that was some sample placeholder for that section, we will renew it. Thanks!

brandonfoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+sets the `isExit = true`, resulting in `Finux` proceeding to call `end()` to exit the Application.
+
+### 3.5 Command Component
+![CommandClassDiagram](img/CommandClassDiagram.png)
```
Would it be better to make this diagram bigger?

brandonfoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+`return -i 2 -d 2021-03-16`
+
+Output:
+Loan marked as returned: [L][2021-03-16] Loan to Tom [v]
```
Should this be in a code block?

marklowsk replied:
I think that was some sample placeholder for that section, we will renew it. Thanks!

brandonfoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+following:
+* negative numbers
+* numbers that refer to non-existent records
+* non-numerics (e.g. alphabets, symbols, etc.)\
```
Seems like there's a small formatting bug here (extra `\`)

8kdesign commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+   1. Valid options: options only for each `Command`. E.g. `-a` for the amount `add` is valid.
+   2. No duplicate options: options are not repeated. E.g. `-a 200 -a 200` or `-l -l` is invalid.
+   3. No conflict options: mutually exclusive options, options that cannot be input at the same time.
+      E.g. ViewCommand implements `{-e | -l | -s}` options, `view -s -l` is considered a conflict of options.
```
Perhaps you might want to explain what {-e | -l | -s} means. I noticed it was explained in your user guide but not here.

marklowsk replied:
Thanks for pointing this out! Will move it over.

aliciatay-zls commented on src/main/java/seedu/duke/command/Command.java:
```diff
@@ -0,0 +1,9 @@
+import seedu.duke.storage.Storage;
+import seedu.duke.ui.Ui;
+
+public abstract class Command {
```
![image](https://user-images.githubusercontent.com/61925875/113077380-9fe3dd80-9203-11eb-8139-1b18be9d9fd7.png)

Could this be hard to read/understand? Maybe just zoom in on one of the classes as an example e.g. AddCommand, since the purpose is just to describe that all these command classes have an attribute COMMAND_*?

marklowsk replied:
Thanks for pointing this out! PlantUML limitations, decided to split the diagram into 2 images.

8kdesign commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+`return -i 2 -d 2021-03-16`
+
+Output:
+Loan marked as returned: [L][2021-03-16] Loan to Tom [v]
```
I think you forgot the inline code here.

marklowsk replied:
I think that was some sample placeholder for that section, we will renew it. Thanks!

isaharon commented on src/main/java/seedu/duke/Finux.java:
```diff
@@ -0,0 +1,85 @@
+     */
+    private void commandLooper() {
+        while (!commandHandler.isExit()) {
+            String rawInput = ui.getUserInput();
```
1. Consider to add the return type `String` for this like the other parts below for consistency.
2. Method arguments need to be updated to match implementation. similarly for `execute()`.
![Presentation1](https://user-images.githubusercontent.com/19249948/113077719-4334f280-9204-11eb-9342-3661f823de6b.png)

marklowsk replied:
Thanks for pointing this out! Will rectify the errors.
We are still in the process of standardizing our diagrams.

brandonfoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+5. Finally, after the extraction to ArrayList<String> is complete, `extractFinalPart` will 
+   call `checkFirstBlock` for the final check to parse any `help` or `creditscore` in the first argument block.
+   
+#### Parser Component Design Consideration
```
Would it be better to include the design considerations under implementation, rather than design? I believe that design is a high-level overview of what each component does, while implementation is for the finer details.

brandonfoong reviewed (commented):
Overall, very good DG - most parts of the code are well-documented and the design choices are explained well. I have left some comments on the formatting and diagrams, but other than those, looks good!

8kdesign commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+### A.2 Value proposition
+
+#### A.2.1 Problem Identification
+Problems faced by students that Finux aim to assist with.
+> * [Problem] Wastage of time due to connection/latency issues went accessing finance tracking website.
+> * [Solution] Use a local based application.
+
+> * [Problem] Hassle to use different applications to keep track of various stuff.
+> * [Solution] Provide an integrated platform to record any expenses, loans, or saving.
+
+> * [Problem] Disorganization issues arising from keep multiple tracking applications.
+> * [Solution] Provide the option to view each category of records within an integrated platform.
+
+> * [Problem] Spending time to gauge whether to borrow money to a friend.
+> * [Solution] Provide a soft gauge through credit score indicators.
```
I like how you format it in terms of problem  -> solution.

aliciatay-zls commented on src/main/java/seedu/duke/command/RemoveCommand.java:
```diff
@@ -0,0 +1,94 @@
+     */
+    @Override
+    public void execute(RecordList recordList, Ui ui, Storage storage, CreditScoreMap creditScoreMap) {
+        Record currentRecord = recordList.getRecordAt(recordNumberInt);
```
![image](https://user-images.githubusercontent.com/61925875/113077995-d5d59180-9204-11eb-8978-93ad35429bc5.png)

Should there currentRecord:Record returned after the call to getRecordAt()?
Is it intentional that the parameters (recordList, creditScoreMap) in the saveData() call are omitted? It might be helpful to see these parameters.

marklowsk replied:
Thanks for pointing this out! Will rectify the errors.
We are still in the process of standardizing our diagrams.

8kdesign reviewed (commented):
Great job! Looking forward to seeing the completed version.

aliciatay-zls reviewed (commented):
Diagrams are very detailed though sometimes overly so. Nice job!

isaharon commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+Before starting to write any code, we recommend that you have a look at Finux's overall design by reading
+[Finux's Architecture](#31-architecture) section.
+
+## 3. Design
```
Design portion is quite comprehensive and done well. Keep up the good work!

isaharon commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+*Figure 4: **Ui** Class Diagram*
+
+The Ui Component consists of a `Ui` class which handles all user input and system output.
+The Ui is only dependent on the `Duke` class and does not interact directly with other classes,
```
Minor typo for `Duke`. Make sure to check for it e.g. `Setting Up` section in locating `Duke.java`.

isaharon commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,630 @@
+
+The following sections below will provide more details of each component.
+
+### 3.2 UI Component
```
Might want to add subheading `Description` for consistency with the other components.

marklowsk replied:
hmmm. Will discuss with my teammates.

isaharon reviewed (commented):
Looks good! Quite readable from the start for the design part although some parts might be too wordy as compared to the rest.

tzexern committed:
Add List All method

tzexern committed:
Update UG for List All method

tzexern committed:
Update DG

Remove Glossary

tzexern committed:
Merge pull request #165 from marklowsk/branch-UpdateDG-2

Fix diagram errors and add more descriptions to DG

tzexern committed:
Update error message

LeeHanYongAndy committed:
Merge pull request #190 from tzexern/branch-list-v2.1

Add list -a

LeeHanYongAndy committed:
Update view for v2.1

LeeHanYongAndy committed:
update UG for view section

LeeHanYongAndy committed:
update junit test for viewCommandTest

tzexern committed:
Merge pull request #191 from LeeHanYongAndy/branch-commandViewV2.1

Update view for v2.1

LeeHanYongAndy committed:
update UG with v2.1 hyperlink and explanation

add explanation for valid option syntax under 2.2

jonahtwl committed:
Completed UML diagrams for Storage

marklowsk committed:
Merge pull request #192 from LeeHanYongAndy/branch-UserGuide

update UG for view section

jonahtwl committed:
Saved down Storage UML

marklowsk committed:
Update DateFormats and ResolverStyle

tzexern committed:
Update DG for manual testing

marklowsk committed:
Update changes in tests.

jonahtwl committed:
Completed DG segment

- Implementation of steps for save function done

LeeHanYongAndy committed:
Merge pull request #193 from marklowsk/branch-UpdateDateFormats

Update DateFormats and ResolverStyle

LeeHanYongAndy committed:
Merge pull request #194 from tzexern/branch-DG-v2.1

Update DG for manual testing

jonahtwl committed:
Merge branch 'master' into jonah-branch-dg

jonahtwl committed:
Completed section 4 for storage features

jonahtwl committed:
Completed DG section for manual testing

gerardtwk committed:
Merge pull request #195 from jonahtwl/jonah-branch-dg

Jonah branch dg

LeeHanYongAndy committed:
Update UG after PE Dry Run

Fixes #185 #186

LeeHanYongAndy committed:
Update validation for amount

Fixes #170 #179

LeeHanYongAndy committed:
update help after PE dry run feed back

Fixes #176

gerardtwk committed:
Merge pull request #196 from LeeHanYongAndy/branch-UserGuide

Update UG after PE Dry Run

jonahtwl committed:
Merge pull request #197 from LeeHanYongAndy/branch-validateAmount

Update validation for amount

marklowsk committed:
Merge pull request #198 from LeeHanYongAndy/branch-commandHelpV2.1

update help after PE dry run feed back

gerardtwk committed:
Fix file writing bug of creditscore feature

1) Add a whitespace char before and after the file delimiter,
    in Storage class line 94
2) Rename FILE_DELIMITER to FILE_DELIMITER_CHAR in Constant class
3) Rename FILE_DELIMITER_REGEX to FILE_DELIMITER_CHAR_REGEX in
    Storage class

marklowsk committed:
Merge pull request #199 from gerardtwk/branch-creditScoreFileWritingBugFix

Fix file writing bug of creditscore feature

LeeHanYongAndy committed:
Add Andy's PPP

marklowsk committed:
Merge pull request #200 from LeeHanYongAndy/branch-PPP

Add Andy's PPP

jonahtwl committed:
Edited creditscore segment

jonahtwl committed:
Updated section 2.2 and 2.4

jonahtwl committed:
Updated creditscore segment

jonahtwl committed:
Changed static to non static

LeeHanYongAndy committed:
Merge pull request #206 from jonahtwl/jonah-ug-issue-fix

Jonah ug issue fix

LeeHanYongAndy committed:
Merge pull request #207 from jonahtwl/jonah-branch-ui-fix

Changed static to non static

marklowsk committed:
Tidy up command.Utils class

Two more functions to check or clean up.

marklowsk committed:
Modify command.Utils class to follow standard

marklowsk committed:
Clean up UtilsTest and add coverage

marklowsk committed:
Use JUnit assertion instead

marklowsk committed:
Improve test branch coverage UtilsTest

marklowsk committed:
Refactor validateArgument in command.Utils class

marklowsk committed:
Add printError function for error messages

jonahtwl committed:
Added CreditScoreCommandTest

- Fixed a minor error in ReturnCommandTest

jonahtwl committed:
Added AddCommandTest

marklowsk committed:
Merge pull request #209 from jonahtwl/jonah-branch-JUnit

Jonah branch j-unit

jonahtwl committed:
Updated PPP

jonahtwl committed:
Updated some bugs for the Commands.

jonahtwl committed:
Checkstyle error fix

marklowsk committed:
Merge pull request #210 from jonahtwl/jonah-branch-ppp

Jonah branch ppp

LeeHanYongAndy committed:
update DG for v2.1

gerardtwk committed:
First draft for PPP

gerardtwk committed:
Rename ppp to github username

gerardtwk committed:
Merge pull request #208 from marklowsk/branch-RefactorAndTest

Add Refactor and Test for command.Utils class

marklowsk committed:
Merge pull request #213 from gerardtwk/branch-ppp

First draft for PPP

marklowsk committed:
Merge branch 'master' into branch-UpdateUiAndMessages

marklowsk committed:
Update diagrams in Command component

marklowsk committed:
Improve description in Command component

marklowsk committed:
Update some typos in UG and DG

marklowsk committed:
Add my PPP and update AboutUs.md

marklowsk committed:
Merge pull request #212 from LeeHanYongAndy/branch-DeveloperGuide

update DG for v2.1

tzexern committed:
Updated output format

tzexern committed:
Update output format

tzexern committed:
Update Junit tests

To align with output format changes

tzexern committed:
Remove outdated screenshots

tzexern committed:
Update output format

tzexern committed:
Update UG for new screenshots

tzexern committed:
Add new screenshots

tzexern committed:
Update AboutUs

tzexern committed:
Add screenshot

tzexern committed:
Update output format

tzexern committed:
Remove unwanted imports

tzexern committed:
Update Junit tests

LeeHanYongAndy committed:
Merge pull request #214 from marklowsk/branch-DocumentationPPPUGAndDG

Add PPP and update UG and DG

tzexern committed:
Merge branch 'master' into branch-list-output-change

LeeHanYongAndy committed:
Merge pull request #215 from tzexern/branch-list-output-change

v2.1 changes

marklowsk committed:
Merge branch 'master' into branch-UpdateUiAndMessages

tzexern committed:
Update incorrect screenshot

tzexern committed:
Add PPP

marklowsk committed:
Merge pull request #216 from tzexern/branch-PPP

Add PPP

LeeHanYongAndy committed:
update UG on help section

marklowsk committed:
Fix some DG related errors

marklowsk committed:
Add Messages.java and refactor magic strings

Refactor magic strings in AddCommand.java

marklowsk committed:
Refactor AddCommand

LeeHanYongAndy committed:
update ParserHandler variable names

Update DG for consistency to new update on ParserHandler

marklowsk committed:
Merge pull request #217 from LeeHanYongAndy/branch-DeveloperGuide

update UG on help section

LeeHanYongAndy committed:
Merge pull request #218 from marklowsk/branch-UpdateUiAndMessages

Modify DG Update Ui and Add Messages.java

marklowsk committed:
Merge pull request #219 from LeeHanYongAndy/branch-parserV2

update ParserHandler variable names

marklowsk committed:
Update add command feature sequence diagram

LeeHanYongAndy committed:
Add ListFeature for DG

Add documentation for List Feature Section in DG

marklowsk committed:
Merge pull request #220 from LeeHanYongAndy/branch-DeveloperGuide

Add ListFeature for DG

jonahtwl committed:
Updated own Sequence Diagrams

LeeHanYongAndy committed:
Merge pull request #221 from jonahtwl/jonah-dg-fixes

Updated own Sequence Diagrams

marklowsk committed:
Merge branch 'master' into branch-DGUpdatesAndCodeRefactor

tzexern committed:
Add sequence diagrams

tzexern committed:
Update DG

Add Return Feature
Update screenshot for Setting Up

tzexern committed:
Merge branch 'master' into branch-dg-update-return

marklowsk committed:
Merge pull request #222 from tzexern/branch-dg-update-return

Update DG

LeeHanYongAndy committed:
update UG DG and PPP format error

tzexern committed:
Merge pull request #223 from LeeHanYongAndy/branch-PPP

update UG DG and PPP format error

marklowsk committed:
Refactor magic string for validateDate()

tzexern committed:
Update Help Screenshot

tzexern committed:
Merge pull request #224 from marklowsk/branch-DGUpdatesAndCodeRefactor

Small DG updates and code refactor

marklowsk committed:
Merge pull request #225 from tzexern/branch-update-screenshot

Update Help Screenshot

LeeHanYongAndy committed:
update UG and help command

Modify all index to record ID.
Modify help command according to new changes in UG.

tzexern committed:
Merge pull request #226 from LeeHanYongAndy/branch-commandHelpV2.1

update UG and help command

LeeHanYongAndy committed:
Update UG after first UG review

jonahtwl committed:
Updated names for creditScoreReturnedLoansMap

jonahtwl committed:
Merge pull request #227 from LeeHanYongAndy/branch-UserGuide

Update UG after first UG review

LeeHanYongAndy committed:
Merge pull request #228 from jonahtwl/jonah-dg-fixes

Updated names for creditScoreReturnedLoansMap

jonahtwl committed:
Refactored magic strings to Constant and Messages class

marklowsk committed:
Add edits to the sequence diagrams

marklowsk committed:
Merge branch 'master' into branch-DGUpdatesAndCodeRefactor

jonahtwl committed:
Updated UML diagrams and PPP

tzexern committed:
Merge pull request #229 from jonahtwl/jonah-storage-refactor

Updated Storage UML Diagram

marklowsk committed:
More edits to increase understandablilty

tzexern committed:
Update method and output

jonahtwl committed:
Fixed some minor changes

tzexern committed:
Merge pull request #230 from jonahtwl/jonah-branch-ug-fixes

Fixed some minor changes

jonahtwl committed:
More minor changes

LeeHanYongAndy committed:
Refactor v2.1

LeeHanYongAndy committed:
Merge pull request #231 from jonahtwl/jonah-branch-storage-fix

More minor changes

jonahtwl committed:
Merge pull request #232 from LeeHanYongAndy/branch-Refactor

Refactor v2.1

jonahtwl committed:
More minor changes to UML diagram

marklowsk committed:
Add some changes to the language

LeeHanYongAndy committed:
Merge pull request #233 from jonahtwl/jonah-branch-storage-fix

More minor changes to UML diagram

gerardtwk committed:
CreditScoreCommand and RecordList developer guide documentation

tzexern committed:
Update DG screenshot

tzexern committed:
Update DG bugs and grammar

LeeHanYongAndy committed:
Merge pull request #234 from marklowsk/branch-DGUpdatesAndCodeRefactor

Update errors in DG and Code

tzexern committed:
Merge branch 'master' into branch-update-screenshot

# Conflicts:
#	docs/DeveloperGuide.md

LeeHanYongAndy committed:
Merge pull request #235 from gerardtwk/branch-developerGuide3

CreditScoreCommand and RecordList developer guide documentation

gerardtwk committed:
Merge pull request #236 from tzexern/branch-update-screenshot

Update DG

LeeHanYongAndy committed:
Update minor DG error

marklowsk committed:
Merge pull request #237 from LeeHanYongAndy/branch-DeveloperGuide

Update minor DG error

marklowsk committed:
Fixing broken link and GitHub Pages friendly

marklowsk committed:
Update repo landing page readme

gerardtwk committed:
Fix a few mistakes in CreditScoreCommand sequence diagram

gerardtwk committed:
Add a note about omitting method parameters for brevity

LeeHanYongAndy committed:
Merge pull request #238 from gerardtwk/branch-creditScoreCommandSeqDiagramFix

Fix a few mistakes in CreditScoreCommand sequence diagram

jonahtwl committed:
Updated UML diagrams

jonahtwl committed:
More updates

jonahtwl committed:
Updated more parts

tzexern committed:
Update UG screenshot

tzexern committed:
Update return sequence diagram

tzexern committed:
Update DG for new component

tzexern committed:
Update PPP

tzexern committed:
Merge pull request #239 from jonahtwl/jonah-branch-dg-fixes

Jonah dg fixes

jonahtwl committed:
Merge pull request #240 from tzexern/branch-update-dg-2

Update DG for new component, UG screenshot, UML minor change

gerardtwk committed:
Minor fix to note section in RecordList class diagram

- Remove unuse constant in Record class

marklowsk committed:
Merge branch 'master' into branch-FinalFixesForV2.1

tzexern committed:
Refactor Magic Strings

marklowsk committed:
Merge pull request #241 from tzexern/branch-refactor-magic-strings

Refactor Magic Strings

marklowsk committed:
Add some JavaDocs for command.Utils class

marklowsk committed:
Merge branch 'master' into branch-FinalFixesForV2.1

marklowsk committed:
Add JavaDocs comments for exceptions

marklowsk committed:
Add some JavaDocs comments for common

marklowsk committed:
Refactor and Add JavaDocs for FinuxLogger

marklowsk committed:
Add missing JavaDoc class description

marklowsk committed:
Fix typo in JavaDocs of some Command classes

marklowsk committed:
Change minor errors and typos in DG

jonahtwl committed:
Fixed minor DG bugs

marklowsk committed:
Fix CI error for missing summary

marklowsk committed:
Merge branch 'master' into branch-FinalFixesForV2.1

gerardtwk committed:
Paraphrase documentation for CreditScore feature

marklowsk committed:
Fix more bugs in DG

jonahtwl committed:
Merge pull request #243 from gerardtwk/branch-developerGuide4

Paraphrase documentation for CreditScore feature

jonahtwl committed:
Merge pull request #244 from gerardtwk/branch-recordListDgMinorFix

Minor fix to note section in RecordList class diagram

marklowsk committed:
Update PPP

marklowsk committed:
Merge branch 'master' into branch-FinalFixesForV2.1

marklowsk committed:
Fix some discrepancies in DG

marklowsk committed:
Add JavaDoc comments for CommandHandler

tzexern committed:
Merge pull request #242 from marklowsk/branch-FinalFixesForV2.1

Branch Fixes for v2.1

LeeHanYongAndy committed:
Set theme jekyll-theme-cayman

tzexern committed:
Update PPP

marklowsk committed:
Merge pull request #245 from tzexern/branch-PPP-2

Update PPP

marklowsk committed:
Test Pagebreaks 1

LeeHanYongAndy committed:
Reduce word limit for PPP to fit Professor Akshay Hard Limit

marklowsk committed:
Test Pagebreaks 2

jonahtwl committed:
Merge pull request #246 from LeeHanYongAndy/branch-PPP

Reduce word limit for PPP to fit Professor Akshay Hard Limit

marklowsk committed:
Test Pagebreaks 3

tzexern committed:
Update PPP

marklowsk committed:
Update PPP to reduce pages

tzexern committed:
Update Sequence diagram

jonahtwl committed:
Updated PPP

marklowsk committed:
Test Pagebreaks 4

jonahtwl committed:
Merge pull request #247 from tzexern/branch-PPP-3

Update PPP

tzexern committed:
Merge pull request #248 from jonahtwl/jonah-branch-ppp-edit

Updated PPP

LeeHanYongAndy committed:
Reduce word limit for PPP 3

tzexern committed:
Merge pull request #249 from marklowsk/branch-UpdatePPP

Update PPP to reduce pages

marklowsk committed:
Test Pagebreaks 5

jonahtwl committed:
Updated PPP again

jonahtwl committed:
Merge pull request #251 from LeeHanYongAndy/branch-PPP2

Reduce word limit for PPP 3

tzexern committed:
Update PPP again

marklowsk committed:
Test Pagebreaks 6

tzexern committed:
Merge pull request #252 from jonahtwl/jonah-branch-ppp-edit

Updated PPP again

marklowsk committed:
Merge pull request #253 from tzexern/branch-PPP-4

Update PPP again

jonahtwl committed:
Jonah PPP page break

tzexern committed:
Update PPP again 2

jonahtwl committed:
Updated PPP 3

jonahtwl committed:
Merge pull request #254 from tzexern/branch-PPP-5

Update PPP again 2

marklowsk committed:
Merge pull request #255 from jonahtwl/jonah-branch-ppp-edit

Updated PPP 3

tzexern committed:
Update format

tzexern committed:
Add page breaks

tzexern committed:
Update DG

tzexern committed:
Add page break DG

tzexern committed:
Update Figure numbering

tzexern committed:
Add page breaks DG

tzexern committed:
Add page break DG

tzexern committed:
Add page breaks DG

tzexern committed:
Add page breaks DG

tzexern committed:
Add page breaks DG

marklowsk committed:
Update bug after last smoke test

marklowsk committed:
Update PPP

jonahtwl committed:
Merge pull request #256 from marklowsk/branch-UpdatePPP

Update Bugs After Last Smoke Test

marklowsk committed:
Reduce word limit for PPP

gerardtwk committed:
Add developer guide documentation for CreditScoreReturnedLoansMap

gerardtwk committed:
Merge branch 'master' into branch-developerGuide4

marklowsk committed:
Update Mark's name

marklowsk committed:
Merge pull request #257 from gerardtwk/branch-developerGuide4

Add developer guide documentation for CreditScoreReturnedLoansMap

LeeHanYongAndy committed:
update DG typo fix

LeeHanYongAndy committed:
update DG typo fix 2

jonahtwl committed:
Merge pull request #258 from LeeHanYongAndy/branch-DeveloperGuide

update DG typo fix

jonahtwl committed:
Update UserGuide.md

changed tag to v2.1

marklowsk committed:
Test pagebreaks again 1

gerardtwk committed:
Java docs for undocumented classes

jonahtwl committed:
Update PUML diagrams

LeeHanYongAndy committed:
Merge pull request #259 from jonahtwl/master

Update PUML diagrams

marklowsk committed:
Test pagebreaks again 2

marklowsk committed:
Test pagebreaks again 3

marklowsk committed:
Test pagebreaks again 4

marklowsk committed:
Test pagebreaks again 5

marklowsk committed:
Test pagebreaks again 6

marklowsk committed:
Test pagebreaks again 7

jonahtwl committed:
Updated PUML diagram

marklowsk committed:
Merge pull request #261 from jonahtwl/master

Updated PUML diagram

marklowsk committed:
Merge branch 'master' into branch-javaDocs2

jonahtwl committed:
Updated PUML diagrams again

marklowsk committed:
Merge pull request #262 from jonahtwl/master

Updated PUML diagrams again

gerardtwk committed:
Resolve JavaDoc style issues

gerardtwk committed:
Merge branch 'branch-javaDocs2' of https://github.com/gerardtwk/tp into branch-javaDocs2

jonahtwl committed:
Update PUML once again

marklowsk committed:
Merge pull request #263 from jonahtwl/master

Update PUML once again

marklowsk committed:
Merge pull request #260 from gerardtwk/branch-javaDocs2

Java docs for undocumented classes

gerardtwk committed:
Final ppp

LeeHanYongAndy committed:
Merge pull request #264 from gerardtwk/branch-ppp2

Final ppp
